<launch>
	<arg name="start_rviz" default="true"/>

	<!-- Start Webcam data streaming node. -->
	<node pkg="sphero_localization" type="webcam_tracking.py" name="webcam_tracking" output="screen" respawn="true">
		<param name="config_path" value="$(find sphero_localization)/config/config.yaml"/>
	</node>

	<!-- Set data streaming frequency parameter for Kalman filter -->
	<param name="/data_stream_freq" value="20"/>

	<!-- Start Kalman filter nodes launcher. -->
	<node pkg="sphero_localization" type="kalman_filter_launch.sh" name="kalman_launcher" output="screen"/>

	<!-- RViz and marker server -->
	<group if="$(arg start_rviz)">
		<node pkg="rviz" type="rviz" name="rviz"  args="-d $(find sphero_utils)/rviz/kalman.rviz"/>
		<node pkg="sphero_utils" type="marker_server.py" name="marker_server"/>
	</group>
</launch>